tools/Session.php

```diff
@@ -4,49 +4,67 @@
 namespace OC_Blog\Tools;
 
 
+
 class Session {
 
 	private static Session $session;
 
 	public function __construct() {
 
 		if(session_status() == PHP_SESSION_NONE){
-
 			session_start();
-
 		}
 	}
 
-	public static function getSession(){
+	/**
+	 * Démarre une session si ce n'est pas déja fait.
+	 *
+	 * @return Session
+	 */
+	public static function getSession(): Session{
 
 		if (!isset(self::$session)){
 			self::$session = new Session();
 		}
 
 		return self::$session;
-
 	}
 
-	public function setKey($key, $value){
-
+	/**
+	 * Ajoute ou modifie une session avec sa valeur.
+	 *
+	 * @param string $key
+	 * @param array $value
+	 */
+	public function setKey(string $key, array $value): void {
 		$_SESSION[$key] = $value;
-
 	}
 
-	public function getKey($key){
+	/**
+	 * Récupère les valeurs d'une session.
+	 *
+	 * @param string $key
+	 *
+	 * @return mixed|null
+	 */
+	public function getKey(string $key) {
 
 		if (isset($_SESSION[$key])){
 			return $_SESSION[$key];
 		}
 		else{
 			return null;
 		}
-
 	}
 
-	public function setValueKey($key, $valueKey, $value){
-
+	/**
+	 * Modifie la valeur d'un tableau d'une session.
+	 *
+	 * @param string $key
+	 * @param string $valueKey
+	 * @param string $value
+	 */
+	public function setValueKey(string $key, string $valueKey, string $value): void {
```

Comment:
Codacy found an issue: [setValueKey accesses the super-global variable $_SESSION.](https://app.codacy.com/manual/PatOpen/OC_Blog/pullRequest?prid=6084628)